packages/idyll-components/src/dynamic.js

```diff
@@ -25,18 +25,34 @@ class Dynamic extends React.PureComponent {
     this.drag(Selection.select(node));
   }
 
-  render() {
-    const { format, value } = this.props;
+  transformValue() {
+    const { format, value, map, min, step } = this.props;
+    const mapOnArray = Array.isArray(map);
+    const mapOnConditions = typeof map === 'string';
+    if (mapOnArray) {
+      const commonDiff = Math.ceil(Math.abs(step));
+      const index = Math.floor((value - min) / commonDiff);
+      return index < map.length ? map[index] : value;
+    }
+    if (mapOnConditions) {
+      return map;
+    }
     const formatter = Format.format(format);
-    return <span className="idyll-dynamic">{formatter(value)}</span>;
+    return formatter(value);
```

Comment:
Since both the `format` property and the new `map` property both specify how to transform the value to be displayed, it may make sense to combine these two into one prop.

What about using `format` for both, and we take different actions depending on whether that is a string or a function:

```
// standard string formatting
[Dynamic value:x format:"d" /] 
```


```
// custom formatting function
[Dynamic value:x format:`(d) => { return ... }` /] 
```

Reply:
Yes, it's better to pass the format function to the same prop. I have updated this so format prop works as described in your comment.